src/main/java/duke/Command.java

```diff
@@ -0,0 +1,148 @@
+package duke;
+import duke.exception.DukeException;
+import duke.tasks.*;
+import duke.controllers.Ui;
+
+import java.time.format.DateTimeParseException;
+
+
+public class Command {
+
+    private String[] cmd;
+    private Ui ui = new Ui();
+    private boolean isExiting = false;
+
+    public Command(String[] cmd) {
+        this.cmd = cmd;
+    }
+
+    /**
+     * Determines the Nature of the command to be executed
+     * @param taskList the TaskList which is active
+     * @param storage The object used to write and save state of Tasklist
+     * @throws DukeException
+     */
+    public String execute(TaskList taskList, Storage storage) {
+        String response = "";
+        try {
+            switch (cmd[0]) {
+            case "list":
+                response = ui.printList(taskList.getList());
+                break;
+            case "mark":
+                response = taskList.mark(cmd[1]);
+                break;
+            case "unmark":
+                response = taskList.unmark(cmd[1]);
+                break;
+            case "find":
+                response = taskList.searchKeyword(cmd[1]);
+                break;
+            //Event creation
+            case "todo":
+                response = createTodo(cmd, taskList);
+                break;
+            case "deadline":
+                response = createDeadline(cmd, taskList);
+                break;
+            case "event":
+                response = createEvent(cmd, taskList);
+                break;
+            case "delete":
+                response = ui.showTaskRemoved(taskList.removeTask(cmd[1]));
+                break;
+            case "bye":
+                isExiting = true;
+                break;
+            default:
+                //TODO: make it a exception
+                throw new DukeException("I'm not sure what that means :((");
+            }} catch (Exception e) {
+                response = e.getMessage();
+            } finally {
+            storage.saveList();
+            return response;
+        }
+    }
+
+
+    public boolean getExitStatus() {
+        return this.isExiting;
+    }
+
+//    public void validateTaskCreation(String[] cmd) {
+//        if (cmd.length != 2) {
+//    }
```

Comment:
Perhaps it would be better to remove this code entirely instead of leaving it commented. I noticed a few other places where quite a large number of unused code lines are commented too.